app/src/main/java/app/insti/fragment/FeedFragment.java

```diff
@@ -122,12 +126,10 @@ public void onFailure(Call<NewsFeedResponse> call, Throwable t) {
         });
     }
 
-    private void displayEvents(final List<Event> events) {
-        /* Skip if we're already destroyed */
-        if (getActivity() == null || getView() == null) return;
-
+    /** Initialize the add event fab if the user has permission */
+    private void initFab() {
         if (((MainActivity) getActivity()).createEventAccess()) {
```

Comment:
Consider moving MainActivity$createEventAccess() to Helpers

Reply:
Will leave this to a later refactor